<!--
    published topics:
    /velodyne_points(sensor_msgs/PointCloud2)
    /velodyne_packets(velodyne_msgs/VelodyneScan)
-->
<launch>
  <!-- declare arguments with default values -->
  <arg name="model" default="64E_S2"/>
  <arg name="pcap" default="" />
  
  <!--  <arg name="calibration" default="$(find velodyne_pointcloud)/params/64e_utexas.yaml"/>-->
  <arg name="calibration" default="$(env HOME)/S2-Unit.yaml"/>  
  <arg name="min_range" default="2.0"/>
  <arg name="max_range" default="250.0"/>
  <arg name="topic_name" default="points_raw"/>

  <!-- start nodelet manager and driver nodelets -->
  <include file="$(find velodyne_driver)/launch/nodelet_manager.launch">
    <arg name="model" value="$(arg model)"/>
    <arg name="pcap" value="$(arg pcap)"/>
  </include>

  <!-- start cloud nodelet -->
  <node pkg="nodelet" type="nodelet" name="velodyne_nodelet"
        args="load velodyne_pointcloud/CloudNodelet velodyne_nodelet_manager">
    <param name="calibration" value="$(arg calibration)"/>
    <param name="min_range" value="$(arg min_range)" />
    <param name="max_range" value="$(arg max_range)" />
    <remap from="velodyne_points" to="$(arg topic_name)"/>
  </node>
</launch>
